assertion/function/assertiontree/util.go

```diff
@@ -61,6 +61,10 @@ func RootNodeSlicesString(name string, rootNodes []*RootAssertionNode) string {
 // this combines each of the special-cased deep nilability introspectors to give a method that
 // determines a deep nilability trigger for an arbitrary assertion node
 func deepNilabilityTriggerOf(node AssertionNode) annotation.ProducingAnnotationTrigger {
+	if node == nil {
```

Comment:
(fix for NilAway reported error that was reported due the support added in this PR)

Reply:
I wonder where `deepNilabilityTriggerOf` ever gets passed a NILABLE node 🤔

Reply:
Never directly. It is a false positive from a context sensitivity point of view: [assertiontree/util.go:L106](https://sourcegraph.uberinternal.com/code.uber.internal/go-code/-/blob/src/code.uber.internal/infra/progsys/nilaway/assertion/function/assertiontree/util.go#L106) sets parent to be nil, and [assertiontree/index_assertion_node.go#L31](https://sourcegraph.uberinternal.com/code.uber.internal/go-code/-/blob/src/code.uber.internal/infra/progsys/nilaway/assertion/function/assertiontree/index_assertion_node.go#L31), for instance, calls `deepNilabilityTriggerOf` with `i.Parent()`, which could be nil from the previous finding.